gunicorn/workers/sync.py

```diff
@@ -186,6 +186,17 @@ def handle_request(self, listener, req, client, addr):
             finally:
                 if hasattr(respiter, "close"):
                     respiter.close()
+            # Discard data in the body before closing the socket to prevent
+            # unexpected TCP RST from being sent.
+            try:
+                discarded_len = req.body.discard()
```

Comment:
It feels like this should be an a `finally` block...but I guess gunicorn doesn't really do request pipelining so it probably doesn't make much difference.